server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -1631,63 +1636,93 @@ void removeEntitlement(Entitlement entitlement, boolean regenModified) {
-        // don't care about updating compliance either.
-        if (regenModified) {
-            // Check consumer's new compliance status and save:
+        log.info("Recomputing status for {} consumers.", distinctConsumers.size());
+        int i = 1;
+        for (Consumer consumer : distinctConsumers) {
+            if (i++ % 1000 == 0) {
+                consumerCurator.flush();
+            }
             complianceRules.getStatus(consumer);
         }
+        consumerCurator.flush();
 
-        sink.queueEvent(event);
+        log.info("All statuses recomputed.");
+
+        // for each deleted entitlement, create an event
+        for (Entitlement entitlement : entsToRevoke) {
+            Consumer consumer = entitlement.getConsumer();
+            Event event = eventFactory.entitlementDeleted(entitlement);
+            if (!entitlement.isValid() &&
+                    entitlement.getPool().isUnmappedGuestPool() &&
+                    consumerCurator.getHost(consumer.getFact("virt.uuid"), consumer.getOwner()) == null) {
+                event = eventFactory.entitlementExpired(entitlement);
+                event.setMessageText(event.getMessageText() + ": " +
+                    i18n.tr("Unmapped guest entitlement expired without establishing a host/guest mapping."));
+            }
+            sink.queueEvent(event);
+        }
+    }
+
+    /**
+     * Filter the given entitlements so that this method returns only
+     * the entitlements that are part of some stack.
+     * @param entitlements Entitlements to be filtered
+     * @return Entitlements that are stacked
+     */
+    private List<Entitlement> filterStackingEntitlements(List<Entitlement> entitlements) {
+        List<Entitlement> stackingEntitlements = new ArrayList<Entitlement>();
+
+        for (Entitlement ent : entitlements) {
+            Pool pool = ent.getPool();
+
+            if (!"true".equals(pool.getAttributeValue("pool_derived")) &&
+                    pool.getProduct().hasAttribute("stacking_id")) {
+                stackingEntitlements.add(ent);
+            }
+        }
+        return stackingEntitlements;
     }
 
     @Override
     @Transactional
     public void revokeEntitlement(Entitlement entitlement) {
-        removeEntitlement(entitlement, true);
+        revokeEntitlements(Collections.singletonList(entitlement));
     }
 
     @Override
     @Transactional
     public int revokeAllEntitlements(Consumer consumer) {
-        int count = 0;
-        for (Entitlement e : entitlementCurator.listByConsumer(consumer)) {
-            removeEntitlement(e, false);
-            count++;
-        }
-        // Rerun compliance after removing all entitlements
-        complianceRules.getStatus(consumer);
-        return count;
+        List<Entitlement> entsToDelete = entitlementCurator.listByConsumer(consumer);
+        revokeEntitlements(entsToDelete);
+        return entsToDelete.size();
```

Comment:
Answer to "why it is not needed any more"

The current code actually computes compliance every time at the end of revokeEntitlements method.

Answer to: "Looks like there were two APIs to remove all entitlements of a consumer, one with compliance check and other without"

I think the one without check was there again for performance reasons. It was there so that its possible to recompute compliance after all entitlements are revoked. This is however exactly what the current batch revoke does.